main.go

```diff
@@ -2,22 +2,65 @@ package main
 
 import (
 	"fmt"
-	"net/http" 
+	"os"
+	"net/http"
+
 	"github.com/gin-gonic/gin"
+	"github.com/joho/godotenv"
+	"gorm.io/driver/postgres"
+  "gorm.io/gorm"
 )
 
+type Todo struct {
+	Id uint
+	Title string
+	Description string
+	Deadline string
+	Status bool
+	Priority int
+}
+
 func main() {
 	router := gin.Default()
-	router.GET("/todo/:id", ShowTodo) 
-
+	router.GET("/todo/:id", func(c *gin.Context) {
+		USER, PASSWORD, PORT, DBNAME :=  GetEnv()
+		postgresPass := "postgresql://" + USER + ":" + PASSWORD + "@db" + ":" + PORT + "/" + DBNAME
+		ShowTodo(c, postgresPass)
+	})
 	router.Run(":8080")
 }
 
-func ShowTodo(c *gin.Context) {
+func ShowTodo(c *gin.Context, postgresPass string) {
 	id := c.Param("id")
-	fmt.Println("id出力", id)
+	
+	db, err := gorm.Open(postgres.Open(postgresPass), &gorm.Config{})
+	if err != nil {
+		errorMessage := err.Error()
+		fmt.Println("DB接続エラー")
+		panic(errorMessage)
+	}
+
+	fmt.Println("DB接続に成功しました。")
+
+	var todo Todo
+  db.First(&todo, id)
 
 	c.JSON(http.StatusOK, gin.H{
 		"message": "ok",
+		"todo": todo,
 	})
 }
+
+func GetEnv() (string, string, string, string) {
```

Comment:
戻り値にerrorを追加して返すのが良さそうです。
```suggestion
func GetEnv() (string, string, string, string, error) {
```
goのエラー処理では関数内でerrorを返すものがある場合、
呼び出し側の関数もerrorを返すことが期待されます。
なので戻り値にerrorを返した上で、

以下のようにするのが良さそうです。

```
	err := godotenv.Load("env/dev.env")
	if err != nil {
		return "", "", "", "", err
	}
```